ConfigSummary.py

```diff
@@ -0,0 +1,368 @@
+        result_parameters = [self.runACEseqQc]
+        if parameters.get(self.runACEseqQc, "false"):
+            result_parameters += [self.HTSLIB,
+                                  self.VCFTOOLS]
+        return subdict_str(parameters, result_parameters)
+
+    def fingerprinting_parameters(self,
+                                  plugin_version: Mapping[str, str],
+                                  parameters: Mapping[str, str]) -> Mapping[str, Optional[str]]:
+        result_parameters = [self.runFingerprinting]
+        if parameters.get(self.runFingerprinting, "false") == "true":
+            result_parameters += [self.fingerPrintingSitesFile]
+        return subdict_str(parameters, result_parameters)
+
+    def coverage_plot_parameters(self,
+                                 plugin_version: Mapping[str, str],
+                                 parameters: Mapping[str, str]) -> Mapping[str, Optional[str]]:
+        result_parameters = [self.runCoveragePlots]
+        if parameters.get(self.runCoveragePlots, "false"):
+            result_parameters += [self.BEDTOOLS]
+        return subdict_str(parameters, result_parameters)
+
+    def summarize(self,
+                  plugin_versions: Mapping[str, str],
+                  parameters: Mapping[str, str]) -> \
+                      Mapping[str, Mapping[str, Optional[str]]]:
+        """
+        This function recapitulates the logic of the workflow. Only parameters and versions are
+        returned that were actually relevant for the analysis described by the parameters and
+        plugin_versions. Therefore, version-specific differences may also be addressed here.
+
+        Returns a nested dictionary structure with string values, that describe the parameters.
+        """
+        # Copy the parameters. ChainMap needs a MutableMapping.
+        mutable_parameters = {k: v for k, v in parameters.items()}
+        # Some parameters may not be set and would otherwise be reported as "null". Their defaults
+        # are generally "false". We fill these values in, in case they are missing.
+        parameters = ChainMap(mutable_parameters, self.possibly_missing_value_defaults)
+        results: Mapping[str, Mapping[str, Optional[str]]] = {
+            "workflow": {
+                "id": parameters[self.WORKFLOW_ID]
+            },
+            "roddy": plugin_versions,
+            "base": subdict_str(parameters,
+                                [self.PERL,
+                                 self.PYTHON,
+                                 self.R,
+                                 self.SAMTOOLS
+                                 ]),
+            "adapter_trimming": self.trimmomatic_parameters(plugin_versions, parameters),
+            "fastqc": self.fastqc_parameters(plugin_versions, parameters),
+            "duplication_marking": self.duplication_parameters(plugin_versions, parameters),
+            "sorting": self.sorting_parameters(plugin_versions, parameters),
```

Comment:
Consider simplifying this further
1. State original variable setting
2. Add "effektive" setting: E.g. in this example: "sort-tool": "samtools sort" or "sort-tool": "biobambam bamsort", or similar.